modules/tile-converter/src/i3s-converter/helpers/geometry-converter.ts

```diff
@@ -738,18 +768,117 @@ function getBatchIds(
 /**
  * Convert GLTF material to I3S material definitions and textures
  * @param sourceMaterials Source GLTF materials
+ * @param shouldMergeMaterials - if true - the converter will try to merge similar materials
+ *                               to be able to merge primitives having those materials
  * @returns Array of Couples I3SMaterialDefinition + texture content
  */
-function convertMaterials(
-  sourceMaterials: GLTFMaterialPostprocessed[] = []
-): I3SMaterialWithTexture[] {
-  const result: I3SMaterialWithTexture[] = [];
+async function convertMaterials(
+  sourceMaterials: GLTFMaterialPostprocessed[] = [],
+  shouldMergeMaterials: boolean
+): Promise<I3SMaterialWithTexture[]> {
+  const materials: I3SMaterialWithTexture[] = [];
   for (const sourceMaterial of sourceMaterials) {
-    result.push(convertMaterial(sourceMaterial));
+    materials.push(convertMaterial(sourceMaterial));
+  }
+
+  if (!shouldMergeMaterials) {
```

Comment:
Thought: Split into convertMaterial and mergeMaterials and call the second conditionally instead of passing option?

Reply:
Done